src/client/dfs/dfs.c

```diff
@@ -3322,15 +3324,47 @@ dfs_obj_set_chunk_size(dfs_t *dfs, dfs_obj_t *obj, int flags, daos_size_t csize)
 		D_GOTO(out, rc = daos_der2errno(rc));
 	}
 
-	/** if this is root object, we need to update the cached handle csize */
-	if (daos_oid_cmp(obj->oid, dfs->root.oid) == 0)
-		dfs->root.d.chunk_size = csize;
-
 out:
 	daos_obj_close(oh, NULL);
 	return rc;
 }
 
+int
+dfs_obj_set_chunk_size(dfs_t *dfs, dfs_obj_t *obj, int flags, daos_size_t csize)
+{
+	int	rc;
+
+	if (obj == NULL)
+		return EINVAL;
+	if (!S_ISDIR(obj->mode))
+		return ENOTSUP;
+	if (csize == 0)
+		csize = dfs->attr.da_chunk_size;
+
+	rc = set_chunk_size(dfs, obj, csize);
+	if (rc)
+		return rc;
+
+	/** if this is the root dir, we need to update the cached handle csize */
+	if (daos_oid_cmp(obj->oid, dfs->root.oid) == 0)
+		dfs->root.d.chunk_size = csize;
+
+	return 0;
+}
+
+int
+dfs_update_chunk_size(dfs_t *dfs, dfs_obj_t *obj, daos_size_t csize)
```

Comment:
Any reason why this doesn't have `obj` in the name? I.e.
```
dfs_obj_set_chunk_size(dfs_t *dfs, dfs_obj_t *obj, int flags, daos_size_t csize)
dfs_update_chunk_size(dfs_t *dfs, dfs_obj_t *obj, daos_size_t csize)
```

Reply:
so the former is for users to set the default chunk size for all files on directory (allowed only on directory objects).
the latter was just meant to be an internal routine to "fix" the chunk size on an existing file. users will not use this api directly as it's only called from the tools/hdlrs. but  i'll rename it in the next patch to the feature branch to something like: dfs_file_update_chunk_size to make it clearer.